components/dialog/dialog-mixin.js

```diff
@@ -476,6 +481,9 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			styles.top = `${iframeTopOverride}px`;
 		}
 
+		if (!this._focusableContentElemPresent) this.shadowRoot.querySelector('.d2l-dialog-content')?.setAttribute('tabindex', '0');
+		else this.shadowRoot.querySelector('.d2l-dialog-content')?.removeAttribute('tabindex');
```

Comment:
Is there a way to avoid manipulating DOM nodes inside of `render()` here? I guess the mixin isn't responsible for rendering the dialog content, but maybe the places that are could either be passed the value of `focusableContenteElemPresent` or they could access it themselves?

Reply:
Yup I think we can do that. The disadvantage is that it becomes another thing that consumers of `DialogMixin` would have to do, and we'd want to update places that use `DialogMixin` (but outside of core there are only 2 so that's not too bad).

Reply:
Cool. I think I'd personally prefer that... it feels weird that the mixin even knows about an element with that CSS class, let alone reaching in to manipulate it.

Reply:
Agreed - I'll make that change.